src/Bridge/Symfony/DependencyInjection/AmenophisElasticsearchExtension.php

```diff
@@ -4,46 +4,88 @@
     {
-        foreach ($mergedConfig['clients'] as $clientName => $clientConfig) {
-            $this->loadClient($clientName, $clientConfig, $container);
-        }
-
         $loader = new XmlFileLoader(
             $container,
             new FileLocator(__DIR__.'/../Resources/config')
         );
         $loader->load('services.xml');
+
+        foreach ($mergedConfig['clients'] as $clientName => $clientConfig) {
+            $this->loadClient($clientName, $clientConfig, $container);
+        }
+
+        foreach ($mergedConfig['indices'] as $alias => $indexConfig) {
+            $mappings = $indexConfig['mappings'];
+            $settings = $indexConfig['settings'];
+            $this->loadIndex($alias, $mappings, $settings, $container);
+        }
     }
 
     private function loadClient(string $clientName, array $config, ContainerBuilder $container)
     {
-        $factory = (new Definition(ClientBuilder::class))
+        $clientFactoryDefinition = (new Definition(ClientBuilder::class))
             ->setFactory([ClientBuilder::class, 'create'])
             ->addMethodCall('setHosts', [$config['hosts']])
             ->setPublic(true)
         ;
-
-        $def = (new Definition(Client::class))
-            ->setFactory([$factory, 'build'])
+        $clientDefinition = (new Definition(Client::class))
+            ->setFactory([$clientFactoryDefinition, 'build'])
             ->setPublic(true)
             ->addTag('amenophis.client', ['key' => $clientName])
         ;
+        $clientServiceId = $this->getClientServiceId($clientName);
+        $container->setDefinition($clientServiceId, $clientDefinition);
+        $container->registerAliasForArgument($clientServiceId, Client::class, $clientName.'Client');
+
+        $indexBuilderDefinition = (new Definition(IndexBuilder::class, [new Reference($clientServiceId)]))
+            ->addTag('amenophis.index_builder', ['key' => $clientName])
+        ;
+        $indexBuilderServiceId = $this->getIndexBuilderServiceId($clientName);
+        $container->setDefinition($indexBuilderServiceId, $indexBuilderDefinition);
+        $container->registerAliasForArgument($indexBuilderServiceId, IndexBuilder::class, $clientName.'IndexBuilder');
+    }
+
+    private function loadIndex(string $alias, ?array $mappings, ?array $settings, ContainerBuilder $container)
+    {
+        $indexDefinition = new Definition(Index::class, [
+            $alias,
+            $mappings,
+            $settings,
+        ]);
+        $indexDefinition->addTag('amenophis.index', ['key' => $alias]);
+
+        $serviceId = $this->getIndexServiceId($alias);
+        $container->setDefinition($serviceId, $indexDefinition);
+        $container->registerAliasForArgument($serviceId, Index::class, $alias.'Index');
+    }
 
-        $serviceId = 'amenophis.client.'.$clientName;
+    private function getClientServiceId(string $clientName): string
+    {
+        return 'amenophis.client.'.$clientName;
```

Comment:
You should consider another prefix like "amenophis_elasticsearch". In case you create another bundle.

Reply:
Done !